src/styles/foundation/_focus-ring.scss

```
@@ -1,18 +1,17 @@
 /// Sets the focus ring for an interactive element
 /// @param {String} $size - The size of the border radius on the focus ring.
 /// @param {String} $style - Focus ring state.
-/// @param {Number} $border-width - The border width of your element in rems.
+/// @param {Number} $border-width - The border width of your element in px.
 ///
 
-@mixin focus-ring($size: 'base', $border-width: rem(0), $style: 'base') {
+@mixin focus-ring($size: 'base', $border-width: '0px', $style: 'base') {
   $stroke: rem(2px);
-  $offset: calc(#{$border-width} + #{rem(1px)});
+  $offset: calc(-1 * calc(#{$border-width} + #{rem(1px)}));
```

@BPScott can you for-see any issues with nesting a calc?

---

Nested calc seems unnecessary, if this is just about OOO, you can wrap it in parens.
```suggestion
  $offset: calc(-1 * (#{$border-width} + #{rem(1px)}));
```

---

It definitely needs to be in a calc because `$border-width` could be a custom property

---

Sorry, this mention totally missed my radar. I'm pretty sure nested calcs are fine.

In the past we've been bitten by [forgetting to wrap math that may involve a custom property in a calc](https://github.com/Shopify/polaris-react/commit/334c3b8a4dc0554f082d58bd15e628923ed1c426#diff-d9cca79bcce0f8de4993d487aa6c3646) and [post-css-env not liking the order of items in a calc when one of the items is a custom property](https://github.com/Shopify/polaris-react/pull/2643) but it doesn't look like this falls into similar traps.

I thiiink Dan might be right, If `$border-width` was a custom property then you'd end up with something like `calc(-1 * (var(--something) + 1px));` looks like legit code - the `var()` is already inside a `calc()`. (Disclaimer this is me assuming things, I've not ran this to confirm)

If you want a bit of extra confidence that we won't be caught by surprise like those last two issues, you can spin up a CRA app and copy polaris's compiled css (the generated `styles.css` you get from doing a `yarn build`) into it and see if it builds without issue.